Fails with java 19

Hi Nick :)

Hope all is well!

I just tried vdmcheck and it fails for java 19. It works with java 17.
In java 19 I get this error: 
```
$ java -jar vdmcheck2.jar ~/source/bitbucket_ts/appfmu/appfmu/modelDescription.xml
Checking XML
Exception in thread "main" java.lang.IncompatibleClassChangeError: class com.fujitsu.vdmj.scheduler.SchedulableThread overrides final method java.lang.Thread.isVirtual()Z
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1013)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1013)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:862)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:760)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:681)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:639)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at com.fujitsu.vdmj.VDMSL.getInterpreter(VDMSL.java:318)
        at com.fujitsu.vdmj.VDMSL.interpret(VDMSL.java:228)
        at com.fujitsu.vdmj.VDMJ.main(VDMJ.java:429)
```

The vdmcheck2.sh script does not seem to work in general.
For java 19: 
```
./VDMCheck2.sh ~/source/bitbucket_ts/appfmu/appfmu/modelDescription.xml
Checking XML
Error: Could not find or load main class com.fujitsu.vdmj.VDMJ
Caused by: java.lang.ClassNotFoundException: com.fujitsu.vdmj.VDMJ
```
For java 17:
```
./VDMCheck2.sh ~/source/bitbucket_ts/appfmu/appfmu/modelDescription.xml
Checking XML
Error: Could not find or load main class com.fujitsu.vdmj.VDMJ
Caused by: java.lang.ClassNotFoundException: com.fujitsu.vdmj.VDMJ
```

Cheers,
Casper

Hi.
Awesome @nickbattle @lausdahl - quick reaction as always ;)
vdmcheck2.jar works on java 19 now.

same story with the shell script:
```
Checking XML
Error: Could not find or load main class com.fujitsu.vdmj.VDMJ
Caused by: java.lang.ClassNotFoundException: com.fujitsu.vdmj.VDMJ
```

Can you make a version that fixes my modelDescription file as well? :D

Windows and a single xml file.

Firstly, I forgot to pass -x. But I have corrected that.


**Mime-type is lost**
These lines https://github.com/INTO-CPS-Association/FMI-VDM-Model/blob/development/fmi2/vdmcheck/src/main/scripts/VDMCheck2.sh#L69-L71
 makes the mime type of INXML text/plain. 
The original mime type of my file is text/xml. 

Makes a file with mime type text/plain.

I wrestled with it a bit, and I end up with the same error:
```
Checking XML
++ which ./VDMCheck2.sh
+ path=/c/Users/caemh/Downloads/vdmcheck-1.1.3-SNAPSHOT/VDMCheck2.sh
++ dirname /c/Users/caemh/Downloads/vdmcheck-1.1.3-SNAPSHOT/VDMCheck2.sh
+ dir=/c/Users/caemh/Downloads/vdmcheck-1.1.3-SNAPSHOT
+ cd /c/Users/caemh/Downloads/vdmcheck-1.1.3-SNAPSHOT
+ VAR=model2205
+ INXSD=schema/fmi2.xsd
+ type java
+ java -jar xsd2vdm.jar -xsd schema/fmi2.xsd -xml /tmp/xml2205.xml -name model2205 -vdm /tmp/vdm2205.vdmsl -nowarn
++ basename /tmp/xml2205.xml
+ BASE=xml2205.xml
+ sed -i -e s+xml2205.xml+XML+g /tmp/vdm2205.vdmsl
+ java -Xmx1g -cp vdmj.jar:annotations.jar com.fujitsu.vdmj.VDMJ -vdmsl -q -annotations -e 'isValidFMIConfiguration(model2205)' model /tmp/vdm2205.vdmsl
+ awk '/^true$/{ print "No errors found."; exit 0 };/^false$/{ print "Errors found."; exit 1 };{ print }'
Error: Could not find or load main class com.fujitsu.vdmj.VDMJ
Caused by: java.lang.ClassNotFoundException: com.fujitsu.vdmj.VDMJ
+ RET=0
+ '[' '' ']'
+ return 0
+ check /tmp/modelDescription2205.xml modelDescription.xml
+ '[' '!' -e /tmp/modelDescription2205.xml ']'
+ return 0
+ check /tmp/buildDescription2205.xml source/buildDescription.xml
+ '[' '!' -e /tmp/buildDescription2205.xml ']'
+ return 0
+ check /tmp/terminalsAndIcons2205.xml icon/terminalsAndIcons.xml
+ '[' '!' -e /tmp/terminalsAndIcons2205.xml ']'
+ return 0
+ exit 0
+ rm -f /tmp/modelDescription2205.xml /tmp/buildDescription2205.xml /tmp/terminalsAndIcons2205.xml /tmp/xml2205.xml /tmp/vdm2205.vdmsl
```

--- a/fmi2/vdmcheck/src/main/scripts/VDMCheck2.sh
+++ b/fmi2/vdmcheck/src/main/scripts/VDMCheck2.sh
@@ -161,7 +161,13 @@ function check()	# $1 = the XML temp file to check, $2 = name of the file
 		BASE=$(basename $1)
 		sed -i -e "s+$BASE+$2+g" "$VDM"
 
-		java -Xmx1g -cp vdmj.jar:annotations.jar com.fujitsu.vdmj.VDMJ \
+		# Fix Class Path Separator - Default to colon for Unix-like systems, , semicolon for Windows
+		CLASSPATH_SEPARATOR=":"
+		if [[ "$OSTYPE" == "msys" || "$OSTYPE" == "win32" ]]; then
+			CLASSPATH_SEPARATOR="\;"
+		fi
+
+		java -Xmx1g -cp vdmj.jar${CLASSPATH_SEPARATOR}annotations.jar com.fujitsu.vdmj.VDMJ \
 			-vdmsl -q -annotations -e "isValidFMIConfiguration($VAR)" \
 			model $VDM |
 			awk '/^true$/{ print "No errors found."; exit 0 };/^false$/{ print "Errors found."; exit 1 };{ print }'

--- a/fmi2/vdmcheck/src/main/scripts/VDMCheck2.sh
+++ b/fmi2/vdmcheck/src/main/scripts/VDMCheck2.sh
@@ -161,10 +161,10 @@ function check()	# $1 = the XML temp file to check, $2 = name of the file
 		BASE=$(basename $1)
 		sed -i -e "s+$BASE+$2+g" "$VDM"
 
-		# Fix Class Path Separator - Default to colon for Unix-like systems, , semicolon for Windows
+		# Fix Class Path Separator - Default to colon for Unix-like systems, , semicolon for msys
 		CLASSPATH_SEPARATOR=":"
-		if [[ "$OSTYPE" == "msys" || "$OSTYPE" == "win32" ]]; then
-			CLASSPATH_SEPARATOR="\;"
+		if [ $OSTYPE == "msys" ]; then
+			CLASSPATH_SEPARATOR=";"
 		fi
 
 		java -Xmx1g -cp vdmj.jar${CLASSPATH_SEPARATOR}annotations.jar com.fujitsu.vdmj.VDMJ \

